docs/src/iteration.md

```diff
@@ -1,31 +1,105 @@
+TGT
+
+julia> only(FwDNAMers{3}("UGU"))
+ERROR:
+[...]
+```
+
+The following kmer iterators are implemented:
+
+### `FwKmers`
+The most basic kmer iterator is `FwKmers`, which simply iterates every kmer, in order:
+
+```@docs
+FwKmers
+FwDNAMers
+FwRNAMers
+FwAAMers
+```
+
+### `FwRvIterator`
+This iterates over a nucleic acid sequence. For every kmer it encounters, it outputs the kmer and its reverse complement.
+
+```@docs
+FwRvIterator
+```
+
+### `CanonicalKmers`
+This iterator is similar to [`FwKmers`](@ref), however, for each `Kmer` encountered, it returns the _canonical_ kmer.
+
+The canonical kmer is defined as the lexographically smaller of a kmer and its reverse complement.
+That is, if [`FwKmers`](@ref) would iterate `TCAC`, then [`CanonicalKmers`](@ref) would return `GTGA`, as this is the reverse complement of `TCAC`, and is before `TCAC` in the alphabet.
+
+[`CanonicalKmers`](@ref) is useful for summarizing the kmer composition of sequences whose strandedness is unknown.
+
+```@docs
+CanonicalKmers
+CanonicalDNAMers
+CanonicalRNAMers
+```
+
+### `UnambiguousKmers`
+[`UnambiguousKmers`](@ref) iterates unambiguous nucleotides (that is, kmers of the alphabets `DNAAlphabet{2}` or `RNAAlphabet{2}`).
+Any kmers containing [ambiguous nucleotides](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC341218/) such as `W` or `N` are skipped.
+
+```@docs
+UnambiguousKmers
+UnambiguousDNAMers
+UnambiguousRNAMers
+```
+
+### `SpacedKmers`
+The [`SpacedKmers`](@ref) iterator iterates kmers with a fixed step size between k-mers.
+For example, for a K of 4, and a step size of 3, the output kmers would overlap with a single nucleotide, like so:
+
+```
+seq: TGATGCGTAGTG
+     TGCT
+        TGCG
+           GTAG
+```
+
+Hence, if `FwKmers` are analogous to `UnitRange`, `SpacedKmers` is analogous to `StepRange`.
 
 ```@docs
-EveryKmer
-EveryKmer{T,S}(seq::S, start::Int = firstindex(seq), stop::Int = lastindex(seq)) where {T<:Kmer,S<:BioSequence}
-EveryKmer(seq::BioSequence{A}, ::Val{K}, start = firstindex(seq), stop = lastindex(seq)) where {A,K}
 SpacedKmers
-SpacedKmers{T,S}(seq::S, step::Int, start::Int, stop::Int) where {T<:Kmer,S<:BioSequence}
-SpacedKmers(seq::BioSequence{A}, ::Val{K}, step::Int, start = firstindex(seq), stop = lastindex(seq)) where {A,K}
-EveryCanonicalKmer
-EveryCanonicalKmer{T}(seq::S, start = firstindex(seq), stop = lastindex(seq)) where {T<:Kmer,S<:BioSequence}
-EveryCanonicalKmer(seq::BioSequence{A}, ::Val{K}, start = firstindex(seq), stop = lastindex(seq)) where {A,K}
-SpacedCanonicalKmers
-SpacedCanonicalKmers{T}(seq::S, step::Int, start = firstindex(seq), stop = lastindex(seq)) where {T<:Kmer,S<:BioSequence}
-SpacedCanonicalKmers(seq::BioSequence{A}, ::Val{K}, step::Int, start = firstindex(seq), stop = lastindex(seq)) where {A,K}
+SpacedDNAMers
+SpacedRNAMers
+SpacedAAMers
+```
+
+The convenience functions [`each_dna_codon`](@ref) and [`each_rna_codon`](@ref) return `SpacedKmers` with a K value of 3 and step size of 3:
```

Comment:
This is very neat! Would it be useful that the `each_dna_codon` method has a `kwarg` for the position or frame?

Reply:
Not really - I think you should simply do `each_codon(@view(seq, 2:end))` instead.